### PR TITLE
Lgtm linter error fixes

### PR DESCRIPTION
**Pull request by LefterisJP**
I used [lgtm.com](https://lgtm.com/projects/g/raiden-network/raiden/) which has already ran a few times in our project. 

- Fixed some minor issues
- Fixed bug in comparison of `ActionCancelRoute`
- Fixed shadowing of variables in inner loops in ec2.py
- Removed unused `maximum_expiration` assignment in `handle_transferrefundcancelroute`

### CODE REVIEW

File: tools/scenario_runner.py [hunk 1]
```diff
@@ -256,8 +254,6 @@ def run(
                 if our_index == 0:
                     last_node = nodes[-1]
                     transfers_by_peer[last_node] = int(amount)
-            else:
```
Comment (palango) [inline tools/scenario_runner.py]: Why is this removed?

Comment (LefterisJP) [reply tools/scenario_runner.py]: `peer` is not used in the else branch or in the code after the `if/else`.
